### PR TITLE
Add paths to make CI strict

### PR DESCRIPTION
**Pull request by F-park**
add `on.(push|pull_request).paths`

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -3,7 +3,19 @@ name: lint
 on:
   push:
     branches: [ master ]
+    paths:
```
Comment (gaogaotiantian) [inline .github/workflows/lint.yml:6]: lint is fast enough that this path restriction is not necessary. It's too wordy and the benefit is just not enough.

File: .github/workflows/documentation-links.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+permissions:
+  pull-requests: write
+
+jobs:
+  documentation-links:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: readthedocs/actions/preview@v1
+        with:
+          project-slug: "viztracer-preview"
```
Comment (gaogaotiantian) [inline .github/workflows/documentation-links.yml:20]: I don't believe this works without the actual readthedocs project.

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -3,7 +3,9 @@ name: docs
 on:
   push:
     branches: [ master ]
-  pull_request:
+    paths:
```
Comment (gaogaotiantian) [inline .github/workflows/docs.yml:6]: Leave the `pull_request` as it is because the readthedocs project is not set up for build & preview. Try to finish one thing for one PR.

**Review by gaogaotiantian — CHANGES_REQUESTED** [review]

File: .github/workflows/documentation-links.yml (hunk 2 above)
Comment (F-park) [reply .github/workflows/documentation-links.yml:20]: I cannot understand, it can work in other [project](https://github.com/dranjan/python-plyfile/blob/master/.github/workflows/pull-request-doc-preview.yml)

Comment (gaogaotiantian) [reply .github/workflows/documentation-links.yml:20]: Yeah, because https://python-plyfile.readthedocs.io/ exists - you need to register the project on readthedocs first to make this work. There's no `viztracer-preview` project on readthedocs so it won't work. Try it locally and see.

Comment (F-park) [reply .github/workflows/documentation-links.yml]: You are right, I should learn how it works and I remove the file.
